src/app/codegen-data-model/tests/TestAttributeReportIBsEncoding.cpp

```diff
@@ -0,0 +1,130 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include "TestAttributeReportIBsEncoding.h"
+
+#include <app/MessageDef/AttributePathIB.h>
+#include <app/MessageDef/AttributeReportIB.h>
+
+using namespace chip::app;
+
+namespace chip {
+namespace Test {
+
+CHIP_ERROR DecodedAttributeData::DecodeFrom(const chip::app::AttributeDataIB::Parser & parser)
+{
+    ReturnErrorOnFailure(parser.GetDataVersion(&dataVersion));
+
+    AttributePathIB::Parser pathParser;
+    ReturnErrorOnFailure(parser.GetPath(&pathParser));
+    ReturnErrorOnFailure(pathParser.GetConcreteAttributePath(attributePath, AttributePathIB::ValidateIdRanges::kNo));
+    ReturnErrorOnFailure(parser.GetData(&dataReader));
+
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR DecodeAttributeReportIBs(ByteSpan data, std::vector<DecodedAttributeData> & decoded_items)
+{
+    // Espected data format:
+    //   CONTAINER (anonymous)
+    //     0x01 => Array (i.e. report data ib)
+    //       ReportIB*
+    //
+    // Overally this is VERY hard to process ...
```

Comment:
"Overally"?

Reply:
I think my brain tried to type `Generally` and `Overall` at the same time... amazing.
Updated.